### PR TITLE
test(client): Regression test for gh-9007

### LINKED ISSUE #9007: Client shows contains filter on uuid data type despite not being a valid operation
### Bug description

When trying to query a relation via a uuid data type I get this error: 
```
/Users/emilymorgan/Desktop/bug/node_modules/@prisma/client/runtime/index.js:36067
          throw new import_engine_core3.PrismaClientUnknownRequestError(message, this.prisma._clientVersion);
                ^

PrismaClientUnknownRequestError3 [PrismaClientUnknownRequestError]: 
Invalid `prisma.coursesOnUsers.findMany()` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("42883"), message: "operator does not exist: uuid ~~* unknown", detail: None, hint: Some("No operator matches the given name and argument type(s). You might need to add explicit type casts."), position: Some(Original(153)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("parse_oper.c"), line: Some(728), routine: Some("op_error") }) }) })
    at cb (/Users/emilymorgan/Desktop/bug/node_modules/@prisma/client/runtime/index.js:36067:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  clientVersion: '2.30.0'
}
```

### How to reproduce

Related to https://github.com/prisma/studio/issues/744

### Expected behavior

For this operator to not be available if it is invalid on a field of type uuid

Screenshot 2021-08-30 at 11 55 12

### Prisma information

Given this schema: 
```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

enum Role {
  STUDENT
  ASSISTANT
  TEACHER
}

enum Visibility {
  PUBLIC
  PRIVATE
}

enum Difficulty {
  EASY
  MEDIUM
  HEAVY
}

model User {
  id String @id @default(uuid()) @db.Uuid

  email    String @db.VarChar
  password String @db.VarChar

  role    Role             @default(STUDENT)
  courses CoursesOnUsers[]
}

model CoursesOnUsers {
  user   User   @relation(fields: [userId], references: [id])
  userId String @db.Uuid

  course   Course @relation(fields: [courseId], references: [id])
  courseId String @db.Uuid

  @@id([userId, courseId])
}

model Course {
  id   String @id @default(uuid()) @db.Uuid
  name String @db.VarChar

  visibility Visibility        @default(PRIVATE)
  users      CoursesOnUsers[]
  topics     TopicsOnCourses[]
}

model TopicsOnCourses {
  topic   Topic  @relation(fields: [topicId], references: [id])
  topicId String @db.Uuid

  course   Course @relation(fields: [courseId], references: [id])
  courseId String @db.Uuid

  @@id([topicId, courseId])
}

model Topic {
  id   String @id @default(uuid()) @db.Uuid
  name String @db.VarChar

  visibility Visibility @default(PRIVATE)

  courses TopicsOnCourses[]
  tasks   TasksOnTopics[]
}

model TasksOnTopics {
  task   Task   @relation(fields: [taskId], references: [id])
  taskId String @db.Uuid

  topic   Topic  @relation(fields: [topicId], references: [id])
  topicId String @db.Uuid

  @@id([taskId, topicId])
}

model Task {
  id   String @id @default(uuid()) @db.Uuid
  name String @db.VarChar

  difficulty Difficulty @default(MEDIUM)
  visibility Visibility @default(PRIVATE)

  topics TasksOnTopics[]
}
```

I ran migrate. 

I added some seeded data - two users and a course. 

Then I ran `npx prisma generate` in a TS project and run the following query:

```
  const data = await prisma.coursesOnUsers.findMany({
    where: {
      userId: {
        contains: "some-valid-uuid",
        mode: "insensitive",
      },
    },
  });
```

### Environment & setup

- OS:  Mac OS
- Database: PostgreSQL
- Node.js version: 16.6

### Prisma Version

```
Environment variables loaded from .env
prisma                : 2.30.0
@prisma/client        : 2.30.0
Current platform      : darwin
Query Engine (Binary) : query-engine 60b19f4a1de4fe95741da371b4c44a92f4d1adcb (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine      : migration-engine-cli 60b19f4a1de4fe95741da371b4c44a92f4d1adcb (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine  : introspection-core 60b19f4a1de4fe95741da371b4c44a92f4d1adcb (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary         : prisma-fmt 60b19f4a1de4fe95741da371b4c44a92f4d1adcb (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash  : 60b19f4a1de4fe95741da371b4c44a92f4d1adcb
Studio                : 0.422.0
```

### PR DESCRIPTION
**Pull request by danstarns**
Closes: https://github.com/prisma/prisma/issues/9007

Related: https://github.com/prisma/prisma-engines/pull/3178

### CODE REVIEW

File: packages/client/tests/functional/issues/9007/tests.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        prisma.user.findMany({
+          where: {
+            // @ts-expect-error
+            uuid: { contains: 'foo-bar' },
+          },
+        }),
+      ).rejects.toThrowError('sm')
+    })
+
+    test('should not generate the contains field on the where type', async () => {
```
Comment (danstarns) [inline packages/client/tests/functional/issues/9007/tests.ts:26]: This test passes in CI, as we no longer generate the contains type for uuid.

File: packages/client/tests/functional/issues/9007/tests.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      await prisma.user.create({ data: {} })
+
+      await expect(() =>
+        prisma.user.findMany({
+          where: {
+            // @ts-expect-error
+            uuid: { contains: 'foo-bar' },
+          },
+        }),
+      ).rejects.toThrowError('sm')
```
Comment (danstarns) [inline packages/client/tests/functional/issues/9007/tests.ts:23]:
```suggestion
      ).rejects.toThrowError()
```

File: packages/client/tests/functional/issues/9007/tests.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      await prisma.user.create({ data: {} })
+
+      await expect(() =>
+        prisma.user.findMany({
+          where: {
+            // @ts-expect-error
+            uuid: { contains: 'foo-bar' },
+          },
+        }),
+      ).rejects.toThrowError()
```
Comment (danstarns) [inline packages/client/tests/functional/issues/9007/tests.ts:23]: **This does not throw an error.** 

We should be throwing a runtime validation error here.

File: packages/client/tests/functional/issues/9007/tests.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+// https://github.com/prisma/prisma/issues/9007
+testMatrix.setupTestSuite(
+  () => {
+    test('should throw an error if using contains filter on uuid type', async () => {
```
Comment (danstarns) [inline packages/client/tests/functional/issues/9007/tests.ts:13]: This fails.

**Comment by tomhoule** [comment]
Persisting findings on the test failures here for context:

There are two things making the tests fail:

 -   Older version of engines. Merging main into the branch fixed it.
 -   Cockroachdb. The fix was only implemented for postgres, and I don't know if LIKE queries on uuids work on cockroach, so I'll have to figure that out and open a PR if they don't.

**Comment by tomhoule** [comment]
Now we know: 


Engines PR coming very soon.

**Comment by tomhoule** [comment]
Relevant: https://github.com/prisma/prisma-engines/pull/3225

**Review by millsp — APPROVED** [review]

**Pull request merged**